traffic_ops/testing/api/v3/servers_test.go

```diff
@@ -26,1017 +26,562 @@ import (
+				}
+			}
+		}
 	}
 }
 
-func GetTestServersIMSAfterChange(t *testing.T, header http.Header) {
-	params := url.Values{}
-	for _, server := range testData.Servers {
-		if server.HostName == nil {
-			t.Errorf("found server with nil hostname: %+v", server)
-			continue
-		}
-		params.Set("hostName", *server.HostName)
-		_, reqInf, err := TOSession.GetServersWithHdr(&params, header)
-		if err != nil {
-			t.Fatalf("Expected no error, but got %v", err.Error())
-		}
-		if reqInf.StatusCode != http.StatusOK {
-			t.Fatalf("Expected 200 status code, got %v", reqInf.StatusCode)
-		}
-	}
-	currentTime := time.Now().UTC()
-	currentTime = currentTime.Add(1 * time.Second)
-	timeStr := currentTime.Format(time.RFC1123)
-	header.Set(rfc.IfModifiedSince, timeStr)
-	for _, server := range testData.Servers {
-		if server.HostName == nil {
-			t.Errorf("found server with nil hostname: %+v", server)
-			continue
-		}
-		params.Set("hostName", *server.HostName)
-		_, reqInf, err := TOSession.GetServersWithHdr(&params, header)
-		if err != nil {
-			t.Fatalf("Expected no error, but got %v", err.Error())
-		}
-		if reqInf.StatusCode != http.StatusNotModified {
-			t.Fatalf("Expected 304 status code, got %v", reqInf.StatusCode)
-		}
+func validateServerFieldsForUpdate(hostName string, expectedResp map[string]interface{}) utils.CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, _ interface{}, _ tc.Alerts, _ error) {
+		params := url.Values{}
+		params.Set("hostName", hostName)
+		servers, _, err := TOSession.GetServersWithHdr(&params, nil)
+		assert.NoError(t, err, "Error getting Server: %v - alerts: %+v", err, servers.Alerts)
+		assert.Equal(t, 1, len(servers.Response), "Expected Server one server returned Got: %d", len(servers.Response))
+		validateServerFields(expectedResp)(t, toclientlib.ReqInf{}, servers.Response, tc.Alerts{}, nil)
 	}
 }
 
-func GetTestServersIMS(t *testing.T) {
-	var header http.Header
-	header = make(map[string][]string)
-	futureTime := time.Now().AddDate(0, 0, 1)
-	timestamp := futureTime.Format(time.RFC1123)
-	header.Set(rfc.IfModifiedSince, timestamp)
-	params := url.Values{}
-	for _, server := range testData.Servers {
-		if server.HostName == nil {
-			t.Errorf("found server with nil hostname: %+v", server)
-			continue
-		}
-		params.Set("hostName", *server.HostName)
-		_, reqInf, err := TOSession.GetServersWithHdr(&params, header)
-		if err != nil {
-			t.Fatalf("Expected no error, but got %v", err.Error())
-		}
-		if reqInf.StatusCode != http.StatusNotModified {
-			t.Fatalf("Expected 304 status code, got %v", reqInf.StatusCode)
+func validateExpectedServers(expectedHostNames []string) utils.CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, resp interface{}, _ tc.Alerts, _ error) {
+		assert.RequireNotNil(t, resp, "Expected response to not be nil.")
+		serverResp := resp.([]tc.ServerV30)
+		var notInResponse []string
+		serverMap := make(map[string]struct{})
```

Comment:
nit: you know exactly how big this map needs to be, so it could be allocated immediately instead of as-needed during iteration of the response.